### PR TITLE
fix: improvements to the test run events

### PR DESCRIPTION
**Pull request by xoscar**
This PR adds several improvements to the events triggered by the BE when executing a test run

## Changes

- Updates event sequence to match the initial vision I never documented 🥴
- Adds better information for the FE when it comes to polling iteration info

## Fixes

- #2400 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## I Encourage you to take a look at the loom video
https://www.loom.com/share/9353520fe43c49c6897f26b45975c2c3

### CODE REVIEW

File: server/executor/assertion_runner.go [hunk 1]
```diff
@@ -208,7 +208,7 @@ func (e *defaultAssertionRunner) emitFailedAssertions(ctx context.Context, req A
 				}
 
 				if errors.Is(spanAssertionResult.CompareErr, expression.ErrInvalidSyntax) {
-					e.eventEmitter.Emit(ctx, events.TestSpecsAssertionError(
```
Comment (xoscar) [inline server/executor/assertion_runner.go:211]: Moving this from errors to warnings as these don't stop the test run

File: server/executor/poller_executor.go [hunk 2]
```diff
@@ -117,50 +115,44 @@ func (pe DefaultPollerExecutor) ExecuteRequest(request *PollingRequest) (bool, s
 				log.Printf("[PollerExecutor] Test %s Run %d: failed to emit TraceDataStoreConnectionInfo event: error: %s\n", request.test.ID, request.run.ID, err.Error())
 			}
 		}
-	}
 
-	err = pe.eventEmitter.Emit(request.ctx, events.TraceFetchingStart(request.test.ID, request.run.ID))
-	if err != nil {
-		log.Printf("[PollerExecutor] Test %s Run %d: failed to emit TraceFetchingStart event: error: %s\n", request.test.ID, request.run.ID, err.Error())
```
Comment (xoscar) [inline server/executor/poller_executor.go:124]: Moving some events ideally this file should only trigger "polling" specific events

File: server/executor/trace_poller.go [hunk 3]
```diff
@@ -141,12 +141,14 @@ func (tp tracePoller) processJob(job PollingRequest) {
 	}
 
 	if job.IsFirstRequest() {
-		err := tp.eventEmitter.Emit(job.ctx, events.TracePollingStart(job.test.ID, job.run.ID))
```
Comment (xoscar) [inline server/executor/trace_poller.go:144]: As main orchestrator, here we should trigger fetching trace events

File: web/src/components/RunEvents/RunEventPolling.tsx [hunk 4]
```diff
@@ -1,31 +1,57 @@
...
+    <S.EventContainer>
+      <S.Dot $logLevel={event.logLevel} />
+      <Typography.Title level={3}>{event.title}</Typography.Title>
+      <Typography.Text>{event.description}</Typography.Text>
+      {!!event.polling && (
+        <S.Content>
+          <S.Column>
+            <S.InfoIcon />
+            <div>
+              <Typography.Title level={3}>Polling profile configuration:</Typography.Title>
```
Comment (xoscar) [inline web/src/components/RunEvents/RunEventPolling.tsx:25]: Adding polling profile info

File: server/executor/poller_executor.go [hunk 5]
```diff
@@ -117,50 +115,44 @@ func (pe DefaultPollerExecutor) ExecuteRequest(request *PollingRequest) (bool, s
...
+		err = pe.eventEmitter.Emit(request.ctx, events.TracePollingStart(request.test.ID, request.run.ID))
+		if err != nil {
+			log.Printf("[PollerExecutor] Test %s Run %d: failed to emit TraceFetchingStart event: error: %s\n", request.test.ID, request.run.ID, err.Error())
```
Comment (danielbdias) [inline server/executor/poller_executor.go:121]:
```suggestion
			log.Printf("[PollerExecutor] Test %s Run %d: failed to emit TracePollingStart event: error: %s\n", request.test.ID, request.run.ID, err.Error())
```

File: server/executor/trace_poller.go [hunk 6]
```diff
@@ -123,7 +123,7 @@ func (tp tracePoller) Stop() {
 func (tp tracePoller) Poll(ctx context.Context, test model.Test, run model.Run) {
 	log.Printf("[TracePoller] Test %s Run %d: Poll\n", test.ID, run.ID)
 
-	job := NewPollingRequest(ctx, test, run, 0)
+	job := NewPollingRequest(ctx, test, run, 1)
```
Comment (danielbdias) [inline server/executor/trace_poller.go:126]: Can we put this number as a constant on this file? Something like `PollingRequestStartIteration` ?

```suggestion
	job := NewPollingRequest(ctx, test, run, PollingRequestStartIteration)
```

**Review by danielbdias — APPROVED** [review]
Great! I've added two suggestions, do they make sense?

**Review by jorgeepc — APPROVED** [review]

**Pull request merged**